### PR TITLE
Improvements for speeding up tests

### PR DESCRIPTION
**Pull request by israelboudoux**
## What does this implement/fix? Explain your changes.
The idea is to build and run only the necessary services for the tests.

## Does this close any currently open issues?
No

## What commands can I run to test the change? 
All the tests used in the pipeline can be simulated locally.

## Any other comments?
No

### CODE REVIEW

File: docker/optimist.Dockerfile [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # build zokrates from source for local verify
-FROM ghcr.io/eyblockchain/local-zokrates:0.8.2 as builder
+FROM docker.io/israelboudoux/tmp-nightfall-zokrates:0.8.2 as builder
```
Comment (druiz0992) [inline docker/optimist.Dockerfile:2]: We should never use things from personal repos.

File: docker/worker.Dockerfile [hunk 2]
```diff
@@ -1,5 +1,5 @@
 # build zokrates from source for local verify
-FROM ghcr.io/eyblockchain/local-zokrates:0.8.2 as builder
+FROM docker.io/israelboudoux/tmp-nightfall-zokrates:0.8.2 as builder
```
Comment (druiz0992) [inline docker/worker.Dockerfile:2]: We should never use assets from personal repos.

File: setup-nightfall [hunk 3]
```diff
@@ -1,4 +1,25 @@
 #! /bin/bash
+
+usage()
+{
+  echo "Usage:"
+  echo "  -g or --ganache; for a ganache simulator"
+}
+
+if [ ! -z "$1" ]; then
+  while [ -n "$1" ]; do
```
Comment (druiz0992) [inline setup-nightfall:10]: Why do we need to force this from the setup phase? We did never require it. What happens if we want to deploy to a different blockchain? If you need to just accelerate test, then create a new setup file with you changes, but do not modify this one

Comment (druiz0992) [inline setup-nightfall:10]: Why do we need to force this from the setup phase? We did never require it. What happens if we want to deploy to a different blockchain? If you need to just accelerate test, then create a new setup file with you changes, but do not modify this one

Comment (israelboudoux) [reply setup-nightfall:10]: This was a mistake and will be removed.

File: docker/optimist.Dockerfile (hunk 1 above)
Comment (israelboudoux) [reply docker/optimist.Dockerfile:2]: Yes, I was just testing, I'll pull from our ECR.

**Comment by signorecello** [comment]
Too bad the gas test is the bottleneck but damn this makes tests much faster

**Review by signorecello — APPROVED** [review]

**Pull request merged**